src/Databases/MySQL/MaterializeMetadata.cpp

```diff
@@ -105,6 +107,47 @@ static Block getShowMasterLogHeader(const String & mysql_version)
     };
 }
 
+static bool rightSyncUserPrivImpl(mysqlxx::PoolWithFailover::Entry & connection, std::ostream & out)
+{
+    Block sync_user_privs_header{
+        {std::make_shared<DataTypeString>(), "current_user_grants"}
+    };
+
+    String grants_query, sub_privs;
+    MySQLBlockInputStream input(connection, "SHOW GRANTS FOR CURRENT_USER();", sync_user_privs_header, DEFAULT_BLOCK_SIZE);
+    while (Block block = input.read())
```

Comment:
What if the user does not have `SELECT` privilege?

Reply:
if the user does not have select priv, it will be catched at https://github.com/ClickHouse/ClickHouse/pull/15977/files/43ca32f753c3d4dd296a44558317c3e8b7309be4#diff-585c4235aec5d560e96d2b24faf3ebd4d14dbbbd5d0425e33defda6f4650abfcR233